src/apify/_actor.py

```diff
@@ -266,12 +267,15 @@ async def finalize() -> None:
         await asyncio.wait_for(finalize(), cleanup_timeout.total_seconds())
         self._is_initialized = False
 
+        # Find out if Scrapy is installed.
+        scrapy_installed = find_spec('scrapy') is not None
+
         if is_running_in_ipython():
             self.log.debug(f'Not calling sys.exit({exit_code}) because Actor is running in IPython')
         elif os.getenv('PYTEST_CURRENT_TEST', default=False):  # noqa: PLW1508
             self.log.debug(f'Not calling sys.exit({exit_code}) because Actor is running in an unit test')
-        elif hasattr(asyncio, '_nest_patched'):
-            self.log.debug(f'Not calling sys.exit({exit_code}) because Actor is running in a nested event loop')
+        elif scrapy_installed:
+            self.log.debug(f'Not calling sys.exit({exit_code}) because Actor is running in Scrapy')
```

Comment:
Well, "scrapy is installed" and "Actor is running in Scrapy" are two different things. I imagine this could cause confusion if somebody run this under a system Python, for example. Isn't there a better way to detect Scrapy?

Reply:
After internal discussion, we decided to utilize `SCRAPY_SETTINGS_MODULE` for that.

Reply:
That, I think, should work.

Reply:
No it won't 😅 Filed https://github.com/apify/apify-sdk-python/issues/401